src/Telegram.Bot.Extensions.Exceptions/Parsing/ApiExceptionDescriptor.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Text.RegularExpressions;
+using Telegram.Bot.Exceptions;
+using Telegram.Bot.Types;
+
+namespace Telegram.Bot.Extensions.Exceptions.Parsing
+{
+    internal class ApiExceptionDescriptor<TException> : IApiExceptionDescriptor
+        where TException : ApiRequestException
+    {
+        public int ErrorCode { get; }
+        public string ErrorMessageRegex { get; }
+        public Type Type => typeof(TException);
+        public ExceptionFactory<TException>? CustomExceptionFactory { get; set; }
+
+        public ApiExceptionDescriptor(int errorCode, string errorMessageRegex)
+        {
+            ErrorCode = errorCode;
+            ErrorMessageRegex = errorMessageRegex;
+        }
+
+        public bool TryParseException(
+            int errorCode,
+            string description,
+            ResponseParameters? responseParameters,
+            out ApiRequestException? exception)
+        {
+            exception = null;
+
+            if (ErrorCode != errorCode || !Regex.IsMatch(description, ErrorMessageRegex))
```

Comment:
We can add a little bit of leeway here by using `RegexOptions.IgnoreCase`

Reply:
Good idea, agree